x-pack/plugins/security_solution/public/flyout/document_details/shared/hooks/use_navigate_to_graph_visualization.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import type { FlyoutPanelProps } from '@kbn/expandable-flyout';
+import { useExpandableFlyoutApi } from '@kbn/expandable-flyout';
+import type { Maybe } from '@kbn/timelines-plugin/common/search_strategy/common';
+import { useKibana } from '../../../../common/lib/kibana';
+import {
+  DocumentDetailsLeftPanelKey,
+  DocumentDetailsRightPanelKey,
+  VisualizationTabGraphKey,
+} from '../constants/panel_keys';
+import { DocumentEventTypes } from '../../../../common/lib/telemetry';
+
+export interface UseNavigateToGraphVisualizationParams {
+  /**
+   * When flyout is already open, call open left panel only
+   * When flyout is not open, open a new flyout
+   */
+  isFlyoutOpen: boolean;
+  /**
+   * Id of the document
+   */
+  eventId: string;
+  /**
+   * Name of the index used in the parent's page
+   */
+  indexName: Maybe<string> | undefined;
+  /**
+   * Scope id of the page
+   */
+  scopeId: string;
+}
+
+export interface UseNavigateToGraphVisualizationResult {
+  /**
+   * Callback to open analyzer in visualize tab
+   */
+  navigateToGraphVisualization: () => void;
+}
+
+/**
+ * Hook that returns a callback to navigate to the graph visualization in the flyout
+ */
+export const useNavigateToGraphVisualization = ({
+  isFlyoutOpen,
+  eventId,
+  indexName,
+  scopeId,
+}: UseNavigateToGraphVisualizationParams): UseNavigateToGraphVisualizationResult => {
+  const { telemetry } = useKibana().services;
+  const { openLeftPanel, openFlyout } = useExpandableFlyoutApi();
+
+  const right: FlyoutPanelProps = useMemo(
+    () => ({
+      id: DocumentDetailsRightPanelKey,
+      params: {
+        id: eventId,
+        indexName,
+        scopeId,
+      },
+    }),
+    [eventId, indexName, scopeId]
+  );
+
+  const left: FlyoutPanelProps = useMemo(
+    () => ({
+      id: DocumentDetailsLeftPanelKey,
+      params: {
+        id: eventId,
+        indexName,
+        scopeId,
+      },
+      path: {
+        tab: 'visualize',
+        subTab: VisualizationTabGraphKey,
```

Comment:
ooooh I get it now! Could you please remove that `VisualizationTabGraphKey` constant from the `panel_keys` file and add a `GRAPH_ID` constant to your `graph_visualization.tsx` file, similar to how we did for [analyzer_graph](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/flyout/document_details/left/components/analyze_graph.tsx#L20) and for [session_view](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/flyout/document_details/left/components/session_view.tsx#L33).

That way we stay consistent with the other 2 tabs